src/main/java/org/dice/factcheckbenchmark/benchmark/rdf/TripleExtractor.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    private void setSimplifiedModel() {
+        simplifiedModel = ModelFactory.createDefaultModel();
+        simplifiedModel.read(new ByteArrayInputStream(simplifiedData.getBytes()), null, "TTL");
+    }
+
+    private void parseStatements() {
+        boolean startNode = false;
+
+        StmtIterator stmtIterator = this.model.listStatements();
+        Resource subjectNode = null;
+        RDFNode objectNode = null;
+        while (stmtIterator.hasNext()) {
+            Statement statement = stmtIterator.next();
+
+            // look for starting node
+            if (statement.getSubject().getURI().matches("^.*__[0-9]*$")) {
+                startNode = true;
+                if (statement.getObject().isResource()) {
+                    subjectNode = statement.getSubject();
+                    if (subjectNode == null)
+                        continue;
+                    processTriple(statement, subjectNode, objectNode);
+                }
+            }
+        }
+
+        if (!startNode) {
+            stmtIterator = this.model.listStatements();
+            subjectNode = null;
+            objectNode = null;
+
+            while (stmtIterator.hasNext()) {
+                Statement statement = stmtIterator.next();
+
+                if (statement.getSubject() != null) {
+                    if (statement.getObject().isResource()) {
+                        subjectNode = statement.getSubject();
+                        if (subjectNode == null)
+                            continue;
+                        processTriple(statement, subjectNode, objectNode);
+                    }
+                }
+            }
+        }
+    }
+
+    private void processTriple(Statement statement, Resource subjectNode, RDFNode objectNode) {
+        this.predicate = statement.getPredicate();
+        objectNode = statement.getObject();
+
+        // check if object is resource and has edges, parse until you get Literal
+        getObject(statement, objectNode);
+
+        // now find if current statement subject node is part of object node
+        // then make that node as subject
+        getSubject(subjectNode);
+    }
+
+
+    /**
+     * retrieve subject node
+     * @param subjectNode subject initial node
+     */
+    private void getSubject(Resource subjectNode) {
+        StmtIterator stmtIterator = this.model.listStatements();
+        while (stmtIterator.hasNext()) {
+            Statement statement = stmtIterator.next();
+
+            if (statement.getObject().isResource()
+                    && statement.getObject().asResource().getURI().equals(subjectNode.getURI())) {
+                subjectNode = statement.getSubject();
+                this.subject = new RDFResource(subjectNode.asResource(), model);
+            }
+        }
+    }
+
+    /**
+     * retrieve object node
+     * @param statement node statement
+     */
+    private RDFNode getObject(Statement statement, RDFNode objectNode) {
```

Comment:
What is the aim of this method? You already have an object node :open_mouth:

Reply:
At this stage, we have ObjectNode that is meaningless to us. Considering FactBench dataset, the data for object/subject nodes are connected via an intermediate node; for example: <http://rdf.freebase.com/ns/m.0jcx__24> dbo:award <http://rdf.freebase.com/ns/m.0dt39>. There it is necessary to parse it for getting meaningful nodes.